app/services/csv_exporter.rb

```diff
@@ -0,0 +1,21 @@
+require 'csv'
+
+module CsvExporter
+  def self.to_csv(resources, columns)
+    Enumerator.new do |csv|
+      csv << csv_header(columns)
+
+      resources.find_each do |obj|
```

Comment:
I've ran into limitations with `find_each` on trying to use this with objects in Katello API controllers. The issue is, find_each is OK when you want to call it on ActiveRecord relations. However some controllers directly return Arrays/Hashes etc.. 

I will check what to do with those so that those controllers can work with this.